src/SignalR/common/Protocols.Json/src/Protocol/JsonHubProtocol.cs

```diff
@@ -500,8 +500,8 @@ private void WriteCompletionMessage(CompletionMessage message, Utf8JsonWriter wr
             }
             else if (message.HasResult)
             {
-                using var token = GetParsedObject(message.Result, message.Result?.GetType());
-                token.RootElement.WriteProperty(ResultPropertyNameBytes.EncodedUtf8Bytes, writer);
+                writer.WritePropertyName(ResultPropertyNameBytes);
+                JsonSerializer.WriteValue(writer, message.Result, message.Result?.GetType(), _payloadSerializerOptions);
```

Comment:
Cool. I didn't realize SignalR also had a use case for this. 👍

Reply:
Perf AF